src/arm7tdmi.rs

```diff
@@ -269,10 +270,22 @@ impl Arm7tdmi {
         self.registers[rd] = op2;
     }
 
-    fn teq(&mut self, rn: u32, op2: u32) {
-        let value = self.registers[rn as usize] ^ op2;
-        self.cpsr.set_sign_flag(value.is_bit_on(31));
-        self.cpsr.set_zero_flag(value == 0);
+    fn tst(&mut self, rn: u32, op2: u32, s: bool) {
+        if s && rn != self.registers[15] {
+            let value = self.registers[rn as usize] & op2;
+            self.cpsr.set_sign_flag(value.is_bit_on(31));
+            self.cpsr.set_zero_flag(value == 0);
+            // Update the C flag during the calculation of Operand2?
```

Comment:
Perfect! 
Since also for the `teq` instruction we have to update the` C` flag, do you prefer that I delete the PR and when the `teq` is fixed, do it again?